homeassistant/components/litejet/light.py

```diff
@@ -1,43 +1,50 @@
 """Support for LiteJet lights."""
 import logging
 
-from homeassistant.components import litejet
 from homeassistant.components.light import (
     ATTR_BRIGHTNESS,
     SUPPORT_BRIGHTNESS,
     LightEntity,
 )
 
+from .const import DOMAIN
+
 _LOGGER = logging.getLogger(__name__)
 
 ATTR_NUMBER = "number"
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up lights for the LiteJet platform."""
-    litejet_ = hass.data["litejet_system"]
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up entry."""
+
+    system = hass.data[DOMAIN]
 
-    devices = []
-    for i in litejet_.loads():
-        name = litejet_.get_load_name(i)
-        if not litejet.is_ignored(hass, name):
-            devices.append(LiteJetLight(hass, litejet_, i, name))
-    add_entities(devices, True)
+    entities = []
+    for i in system.loads():
+        name = await hass.async_add_executor_job(system.get_load_name, i)
+        entities.append(LiteJetLight(config_entry.entry_id, system, i, name))
```

Comment:
We want to minimize the number of jumps into the executor since each job requires pushing it into a queue and waiting for it to finish.  Its quite expensive.

Please move this into its own function and return the entities from it so we only have one jump into the executor

Reply:
Same comment for the other platforms.